### ISSUE TITLE
Invalid URL host: "%s.openai.azure.com"

### ISSUE THREAD

**Issue by shelltea** [carlrobertoh/CodeGPT#449]
### What happened?

<img width="1363" alt="image" src="https://github.com/carlrobertoh/CodeGPT/assets/864375/30814f4b-0fe1-4060-9b08-c284aaecf652">

### Relevant log output or stack trace

```shell
Something went wrong

java.lang.IllegalArgumentException: Invalid URL host: "%s.openai.azure.com"
	at okhttp3.HttpUrl$Builder.parse$okhttp(HttpUrl.kt:1338)
	at okhttp3.HttpUrl$Companion.get(HttpUrl.kt:1634)
	at okhttp3.Request$Builder.url(Request.kt:184)
	at ee.carlrobert.llm.client.azure.AzureClient.buildHttpRequest(AzureClient.java:62)
	at ee.carlrobert.llm.client.azure.AzureClient.getChatCompletionAsync(AzureClient.java:44)
	at ee.carlrobert.codegpt.completions.CompletionRequestService.getChatCompletionAsync(CompletionRequestService.java:82)
	at ee.carlrobert.codegpt.completions.CompletionRequestHandler.startCall(CompletionRequestHandler.java:45)
	at ee.carlrobert.codegpt.completions.CompletionRequestHandler$CompletionRequestWorker.doInBackground(CompletionRequestHandler.java:73)
	at ee.carlrobert.codegpt.completions.CompletionRequestHandler$CompletionRequestWorker.doInBackground(CompletionRequestHandler.java:62)
	at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
```

### Steps to reproduce

1. The configuration information is as follows:

<img width="571" alt="image" src="https://github.com/carlrobertoh/CodeGPT/assets/864375/3cc43b91-9025-4954-887f-af71f6dcf6f2">

2. Whether in the chat window or generating git comments, errors are reported.

### CodeGPT version

2.6.0

### Operating System

macOS

**Comment by carlrobertoh** [comment]
Hi, I recently removed the custom configuration for the Azure service, and because of that, the base host might be incorrect when upgrading from the previous CodeGPT version.

As a workaround, I suggest you reset the settings and try again.

- Find the corresponding CodeGPT configuration files (e.g., `~/.local/share/JetBrains/IntelliJIdea2023.3/options/`).

- Delete the file `CodeGPT_AzureSettings_210.xml`.

**Comment by shelltea** [comment]
Thank you for your reply. Following the method you provided, I deleted CodeGPT_AzureSettings_210.xml and restarted IDEA to reconfigure Azure, but it still shows an error. I noticed that after configuring Azure, it regenerated a file of the same kind.

**Comment by carlrobertoh** [comment]
Another option would be to configure Azure through a custom OpenAI-compatible service. 
I will try to reproduce and fix it in the next release, sorry about that.

<img width="723" alt="Screenshot 2024-04-11 at 14 46 44" src="https://github.com/carlrobertoh/CodeGPT/assets/15891408/f5d4dd7c-90c7-4d96-ab6a-988ac7f8bf9e">

**Comment by shelltea** [comment]
Following the method you provided for using the Custom OpenAI Service, I can now use Azure's OpenAI. Thank you!

**Comment by carlrobertoh** [comment]
This issue should be fixed in the new version 2.6.1. Please reopen it if it is still reproducible.
